types.ts

```diff
@@ -1172,6 +1172,106 @@ export interface RateLimits {
  */
 export type BasicPackDefinition = Omit<PackVersionDefinition, 'version'>;
 
+/**
+ *  Default tool types supported by Coda Packs for job definitions.
+ * @hidden
+ */
+export enum BuiltInToolType {
+  /**
+   * Allows formulas within a pack to be used as tools.
+   */
+  Pack = 'Pack',
+  /**
+   * Allows per pack knowledge to be used as a tool.
+   */
+  PackKnowledge = 'PackKnowledge',
+
+  /**
+   * Allows all ingested pack knowledge to be used as a tool.
+   */
+  GlobalPackKnowledge = 'GlobalPackKnowledge',
+}
+
+/**
+ * The type identifier for a tool
+ * @hidden
+ */
+export type ToolType = keyof ToolMap;
+
+/**
+ * Base interface for all tool definitions.
+ * @hidden
+ */
+export interface BaseTool<T extends string> {
+  /** The type identifier for this tool. */
+  type: T;
+}
+
+/**
+ * Tool that references formulas within a pack.
+ * @hidden
+ */
+export interface PackTool extends BaseTool<BuiltInToolType.Pack> {
+  /** The ID of the pack to use as a tool. */
+  packId: number;
+  /** The specific formulas to make available as tools. */
+  formulas?: Array<{
+    /** The name of the formula to use as a tool. */
+    formulaName: string;
+    /** Optional description for this formula. */
+    description?: string;
+  }>;
+}
+
+/**
+ * Tool that provides access to pack knowledge.
+ * @hidden
+ */
+export interface PackKnowledgeTool extends BaseTool<BuiltInToolType.PackKnowledge> {
+  /** The ID of the pack to use as a tool. */
+  packId: number;
+}
+
+/**
+ * Tool that provides access to all ingested pack knowledge.
+ * @hidden
+ */
+export interface GlobalPackKnowledgeTool extends BaseTool<BuiltInToolType.GlobalPackKnowledge> {}
+
+/**
+ * Map of tool types to their corresponding tool interfaces.
+ * This interface can be extended via declaration merging to add custom tool types.
+ * @hidden
+ */
+export interface ToolMap {
+  [BuiltInToolType.Pack]: PackTool;
+  [BuiltInToolType.PackKnowledge]: PackKnowledgeTool;
+  [BuiltInToolType.GlobalPackKnowledge]: GlobalPackKnowledgeTool;
+}
+
+/**
+ * Union of all supported tool types.
+ * @hidden
+ */
+export type Tool = ToolMap[keyof ToolMap];
+
+/**
+ * Definition of an agent job that can be executed within a pack.
+ * @hidden
+ */
+export interface Job {
+  /** Stable identifier for the agent. */
+  name: string;
+  /** Display name shown to users for this agent. */
+  displayName: string;
+  /** Description of what this agent does. */
+  description: string;
+  /** The prompt/instructions that define the agent's behavior. */
+  prompt: string;
+  /** List of tools that this agent can use. */
```

Comment:
Would be good to comment whether or not this tool's formulas are automatically available or not if not included here?